Tests/MutationTesting/MutationTestingDelegateTests.swift

```diff
@@ -111,4 +111,47 @@ final class MutationTestingDelegateTests: MuterTestCase {
 
         fileManager.changeCurrentDirectoryPath(currentDirectoryPath)
     }
+
+    func test_timeout() async throws {
+        let configuration = MuterConfiguration(
+            executable: "/tmp/swift",
+            arguments: ["test"],
+            testSuiteTimeOut: 9
+        )
+
+        let schemata = try MutationSchema.make(
+            filePath: "/path/fileName",
+            position: .init(line: 1)
+        )
+
+        _ = await sut.runTestSuite(
+            withSchemata: schemata,
+            using: configuration,
+            savingResultsIntoFileNamed: "logFileName"
+        )
+
+        XCTAssertTrue(testingTimeOutExecutor.withTimeLimitCalled)
```

Comment:
What actually causes this test to timeout?

Reply:
the scenario here is that the mutation testing deleate io needs to pass the timeout from the config and call the timeout executor